### ISSUE TITLE
[LogstreamerInput] Does not find some files

### ISSUE THREAD

**Issue by looatgb** [mozilla-services/heka#1618]
LogstreamerInput did not find some of my logs, although they all had the same owner/group/file permissions and followed the same file name pattern.

##### Directory structure

```
/var
`-- /log
    `-- /acme
        `-- machine-1
        `-- machine-2
```

##### Log file names

```
ACME-01-NYC_Main.log-20150301
ACME-01-NYC_Main.log-20150302
ACME-01-NYC_Main.log-20150303
...
ACME-01-NYC_Main.log-20150501
```

##### LogstreamerInput config

```
[AcmeMainLogInput]
type = "LogstreamerInput"
log_directory = "/var/log/acme"
file_match = '(?P<Hostname>[^/]+)/(?P<Device>ACME-\d+-[^_]+)_Main\.log-?(?P<Seq>\d*)'
priority = ["Hostname", "Device", "Seq"]
differentiator = ["Hostname", "_", "Device"]
splitter = "AcmeMainLogSplitter"
decoder = "AcmeMainLogDecoder"
```

When I run `heka-logstreamer --config=/path/to/config.toml` I see only part (not all) Main logs found. None of the DB logs we're found (the DB log differs from Main log only in the suffix, eg. `ACME-01-NYC_DB.log-20150301`). The same for Error logs (none found).

When debugging I even changed the `file_match` regex to `.*`, yet LogstreamerInput could not find these files.

##### Workaround

I've noticed that updateing "invisible" files makes them "visible" to heka again.
I can confirm that running `touch *` in the log directory (which updates the "last modified date" of each file) fixes the problem (all files are visible again).

I believe this is a bug in LogstreamerInput and should be investigated and fixed.

**Comment by looatgb** [comment]
#### More info for debugging

Sorted ASC by filename

Filename | Found? | Last Modified
----------- | ----------- | -----------------
ACME-01-NYC_Main.log-20150419 | Yes | 2015-06-24 17:12
ACME-01-NYC_Main.log-20150426 | Yes | 2015-06-24 17:11
ACME-01-NYC_Main.log-20150503 | No | 2015-05-10 03:46
ACME-01-NYC_Main.log-20150510 | Yes | 2015-06-24 17:13
ACME-01-NYC_Main.log-20150517 | No | 2015-05-24 03:16
ACME-01-NYC_Main.log-20150524 | Yes | 2015-06-25 16:21
ACME-01-NYC_Main.log-20150531 | Yes | 2015-06-25 16:21

**Comment by trink** [comment]
see https://hekad.readthedocs.org/en/latest/config/inputs/logstreamer.html
oldest_duration

**Comment by rafrombrc** [comment]
Yup, the issue here is that the default isn't documented. I've fixed the docs and opened PR #1619.

**Comment by looatgb** [comment]
Thanks. I don't know how I could have missed that. I looked at that doc many times, yet somehow this parameter never made it into my brain.
